FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -211,57 +222,65 @@ extension ComposeViewController {
 
         showSpinner("sending_title".localized)
 
-        Promise<Bool> { [weak self] in
-            try await(self!.encryptAndSendMessage())
-        }.then(on: .main) { [weak self] sent in
-            if sent { // else it must have shown error to user
-                self?.handleSuccessfullySentMessage()
+        switch prepareMessage() {
+        case .failure(let error):
+            switch error {
+            case .internalError(let message):
+                showAlert(message: message)
+            case .missedSender:
+                showAlert(message: "compose_no_pub_sender".localized)
+            case .noPubKeyForRecipient:
+                // already handled
+                break
             }
-        }.catch(on: .main) { [weak self] error in
-            self?.showAlert(error: error, message: "compose_error".localized)
+        case .success(let data):
+            self.messageSender.sendMail(mime: data)
+                .retry(3)
+                .sink{ [weak self] result in
+                    guard case .failure(let error) = result else {
+                        return
+                    }
+                    self?.showAlert(error: error, message: "compose_error".localized)
+                } receiveValue: { [weak self] _ in
+                    self?.handleSuccessfullySentMessage()
+                }
+                .store(in: &cancellable)
```

Comment:
For this demo, please replicate original behavior - without retry or other features not present in the original.

Is the `switch prepareMessage() {` the only way to handle a `Result`? Is there some other possible pattern like:

```swift
let preparedMessage = prepareMessage()
if preparedMessage.isSuccess {

} else {

}
```

?

Reply:
Please check latest commit. Implemented similar to what we have in original code.